### PR TITLE
Move wiki docs into repo

### LINKED ISSUE #112: Move docs into the repository
**Is your feature request related to a problem? Please describe.**

Using GitHub wiki for docs doesn't allow us to have "reviews" in the documentation contribution workflow.

**Describe the solution you'd like**

Move the wiki pages into the repository as markdown files, and create an action that moves text over to the wiki. This means we'll treat the wiki as a read-only space.

This also allows us to have linting and testing to ensure consistency.

**Describe alternatives you've considered**

A docs website is the ideal solution, but is currently not do-able. However, having docs in the repo will allow easier transition to a docs generator if/when that happens.

**Additional context**

This was discussed with @ptomecek and @robambalu in a meeting today.

**Next steps:**

- [ ] Move the documentation **as-is** into the repo
- [ ] Give attribution fairly - add everyone who originally authored the docs as co-authors on the PR
- [ ] Add automation action

### PR DESCRIPTION
**Pull request by pavithraes**
Closes #112 

~~Currently, we can't test the workflow until the PR is merged. However, I think we can temporarily enable runs on PRs to test it.~~

Additional notes added inline.

### CODE REVIEW

File: .github/workflows/publish_wiki.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  workflow_dispatch:
+
+jobs:
+   build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Upload Documentation to Wiki
+        uses: SwiftDocOrg/github-wiki-publish-action@v1
```
Comment (pavithraes) [inline .github/workflows/publish_wiki.yml:18]: Need to verify if we're OK with using this action.

The repository has been archived (hence, unmaintained) - but the action seems fairly stable and works (I've tested it aseparately).

File: docs/wiki/Home.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+# Contributing
+Contributions are welcome on this project. We distribute under the terms of the [Apache 2.0 license](https://github.com/Point72/csp/blob/main/LICENSE).
+
+For **bug reports** or **small feature requests**, please open an issue on our [issues page](https://github.com/Point72/csp/issues).
+
+For **questions** or to discuss **larger changes or features**, please use our [discussions page](https://github.com/Point72/csp/discussions).
+
+For **contributions**, please see our [developer documentation](https://github.com/Point72/csp/wiki/99.-Developer). We have `help wanted` and `good first issue` tags on our issues page, so these are a great place to start.
+
+For **documentation updates**, make PRs that update the pages in `/docs/wiki`. The documentation is pushed to the GitHub wiki automatically through a GitHub workflow. Note that direct updates to this wiki will be overwritten.
```
Comment (pavithraes) [inline docs/wiki/Home.md:34]: This is a new note (i.e., not in the wiki yet) about the docs update+publish workflow.

File: .github/workflows/publish_wiki.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        with:
+          path: "docs/wiki"
+        env:
+          GH_PERSONAL_ACCESS_TOKEN: ${{ secrets.GH_TOKEN }}
```
[the remark this answers is not in the document]
Comment (pavithraes) [reply .github/workflows/publish_wiki.yml:22]: @ngoldbaum - Could you please help verify if `GH_TOKEN` secret have "repo" level scope?

File: .github/workflows/publish_wiki.yml (hunk 1 above)
Comment (pavithraes) [reply .github/workflows/publish_wiki.yml:18]: @timkpaine @trallard - What do you think and prefer here?

Comment (timkpaine) [reply .github/workflows/publish_wiki.yml:18]: @pavithraes action seems fine

**Review by melissawm — APPROVED** [review]
This looks reasonable to me, and we can reevaluate in the future if a new action show up. For now I believe this does the job. Thanks @pavithraes !

**Comment by timkpaine** [comment]
Tested on [my fork](https://github.com/timkpaine/csp)

**Review by timkpaine — APPROVED** [review]

**Review by timkpaine — APPROVED** [review]

**Review by timkpaine — APPROVED** [review]

**Comment by pavithraes** [comment]
@timkpaine Thanks for your updates here - it looks good to me.

File: .github/workflows/publish_wiki.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
```
Comment (melissawm) [inline .github/workflows/publish_wiki.yml:22]: Is there a reason to use v1 instead of [v4](https://github.com/actions/checkout?tab=readme-ov-file#usage)?

**Review by melissawm — APPROVED** [review]
Overall looks good to me so approving, I left a small optional comment. Cheers!

File: .github/workflows/publish_wiki.yml (hunk 4 above)
Comment (timkpaine) [reply .github/workflows/publish_wiki.yml:22]: it should probably be bumped, but dependabot will update it anyway

**Comment by timkpaine** [comment]
https://github.com/Point72/csp/actions/runs/8141618916

image

**Pull request merged**